lib/tasks/message.rake

```diff
@@ -30,4 +30,28 @@ namespace :message do
         puts "#{step.id} #{e.message}"
     end
   end
+
+  desc "Migrate missing gender when start new session"
+  task migrate_missing_gender_on_new_session: :environment do
+    ActiveRecord::Base.record_timestamps = false
+
+    ActiveRecord::Base.transaction do
+      # messenger
+      no_basic_info_text_messages = Message.where(platform_name: "Messenger", province_id: nil, district_id: nil, gender: "")
+      no_basic_info_text_messages.find_each do |message|
+        last_prev_message =  Message.where(content_type: "TextMessage", content_id: message.content_id, status: :completed).where("last_interaction_at < ?", message.created_at)\
```

Comment:
@radinreth create a new method message.last_completed and move this into that

Reply:
Done, Please help me check bong